src/services/motionModel.ts

```diff
@@ -3,21 +3,48 @@ import {
   createMotionModel,
   selectImages,
   packFrameKm,
+  checkFrameKmContinuity,
 } from 'util/motionModel';
-import { GnssMetadata } from 'types/motionModel';
+import { FramesMetadata, GnssMetadata } from 'types/motionModel';
 import { promiseWithTimeout, sleep } from 'util/index';
 import { ifTimeSet } from 'util/lock';
 import { isIntegrityCheckDone } from './integrityCheck';
 import { isCarParkedBasedOnImu } from 'util/imu';
 import { isPrivateZonesInitialised } from './loadPrivacy';
 import { getNextGnss, isGnssEligibleForMotionModel } from 'util/motionModel/gnss';
 import { getNextImu } from 'util/motionModel/imu';
-const ITERATION_DELAY = 10000;
+import { moveFrames } from 'util/frames';
+import { addFramesToFrameKm, clearFrameKmTable, getExistingFramesMetadata, getFrameKmMetadata, getFrameKmName, getFramesCount, isFrameKmComplete } from 'sqlite/framekm';
+import { getConfig } from 'util/motionModel/config';
+import { UNPROCESSED_FRAMEKM_ROOT_FOLDER } from 'config';
+import { join } from 'path';
+import { promises } from 'fs';
 
-export const lastProcessed = null;
+const ITERATION_DELAY = 10000;
+let failuresInARow = 0;
+let endTripTrimmed = true;
 
 const execute = async () => {
   try {
+    const {
+      isTripTrimmingEnabled,
+      TrimDistance,
+      DX
+    } = getConfig();
+
+    if (isTripTrimmingEnabled && !endTripTrimmed) {
+      const data = await getFrameKmMetadata();
```

Comment:
Start trip trimming:

Executes on start, checks the contents of FrameKM table (leftovers of previous drive).
Trims last N records depending on TrimDistance (configurable),
packs and cleans up the table and frame folders.